app/entities/user.go

```diff
@@ -47,24 +48,27 @@ func checkName(name string) error {
 
 func checkMail(email string) error {
 	const (
-		maxLocalEmailLen  int = 64
-		maxDomainEmailLen int = 255
+		maxLocalBytes  int = 64
+		maxDomainBytes int = 255
 	)
-	// Checking the total length (allowed no more than 64+1+255=320 symbols)
-	emailLen := len(email)
-	if emailLen > maxLocalEmailLen+1+maxDomainEmailLen {
-		return errors.New("email contents too many symbols: " + strconv.Itoa(emailLen))
+	// Checking the lengths of local and domain parts
+	atIndex := strings.IndexByte(email, '@')
+	if atIndex > maxLocalBytes {
+		return errors.New("local part of email contains too many bytes: " + strconv.Itoa(atIndex))
 	}
-	// Checking for some email issues by regular expression
-	if valid, _ := regexp.MatchString("(?i)"+`^([^\.@(),:;<>@[\\\]]?[\da-z!#$%&'*+,\-/=?^_\x60{|}~]+.?)+@([^.-][\da-z-]*[^.-]\.?)+([a-z]{2,6})$`, email); !valid {
-		return errors.New("email does not match with regexp \"(?i)\"+`^([^\\.@(),:;<>@[\\\\\\]]?[\\da-z!#$%&'*+,\\-/=?^_\\x60{|}~]+.?)+@([^.-][\\da-z-]*[^.-]\\.?)+([a-z]{2,6})$`")
+	if localPartLen := len(email) - atIndex - 1; localPartLen > maxDomainBytes {
+		return errors.New("domain part of email contains too many bytes: " + strconv.Itoa(localPartLen))
+	}
+	// Checking for other email issues by regular expression
+	if valid, _ := regexp.MatchString("(?i)"+`^(?:[a-z\d!#$%&'*+/=?^_\x60{|}~-]+(?:\.[a-z\d!#$%&'*+/=?^_\x60{|}~-]+)*)@(?:(?:[a-z\d](?:[a-z\d-]*[a-z\d])?\.)+[a-z\d](?:[a-z\d-]*[a-z\d])?)$`, email); !valid {
```

Comment:
btw ignoring errors is not a good practice
```golang
if valid, _ 
```

Reply:
Error processed!

Reply:
Looks neat. We have the same cases in our code, don't we?

Reply:
O chort, sorry, mabut` 4erez github... Bo ja to4no robyv. I will change it soon...